SCClassLibrary/Common/Control/Server.sc

```diff
@@ -348,23 +366,41 @@ Server {
 	}
 
 	initTree {
-		nodeAllocator = NodeIDAllocator(clientID, options.initialNodeID);
-		this.sendMsg("/g_new", 1, 0, 0);
+		this.newNodeAllocators;
+		this.sendMsg("/g_new", this.defaultGroupID, 0, 0);
 		tree.value(this);
 		ServerTree.run(this);
 	}
 
 	/* id allocators */
 
+	// private, called from server notify response with next free clientID
 	clientID_ { |val|
+		var failstr = "Server % couldn't set client id to: % - %.
+clientID is still %.";
+
+		if(val == clientID) {
+			// no need to change
+			^this
+		};
 		if(val.isInteger.not) {
-			"Server % couldn't set client id to: %".format(name, val.asCompileString).warn;
+			failstr.format(name, val.cs, "not an Integer", clientID).warn;
 			^this
 		};
-		if(clientID != val) {
-			clientID = val;
-			this.newAllocators;
-		}
+		if (val < 0) {
+			failstr.format(name, val.cs, "less than minimum 0", clientID).warn;
+			^this
+		};
+		if (val >= this.numClients) {
+			failstr.format(name, val.cs,
+				"greater than server.numClients % - 1 allows".format(this.numClients), clientID
```

Comment:
i know it may not sound like a great idea, but i'd prefer to see multi-line method invocations written with one argument per line. i almost missed "clientID" the first time, then thought it might be part of the format message.

```supercollider
failstr.format(
  name,
  val.cs,
  "...".format(this.numClients),
  clientID
).warn;
```

but up to you. 

other than that i think this is a superbly written method.